# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/nsNSSComponent.h
# Commit: 889d7cd14e4d
# Full Hash: 889d7cd14e4d76462eabe4a26a551c72fb416e86
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2020-06-05 14:35:19
# Regressor Bug: 1630434
# File Overlap Count: 1
# Description:
#   Bug 1630434 - de-duplicate preloaded intermediates that may have been cached in cert9.db r=kjacobs,bbeurdouche
#   
#   In general, PSM caches intermediates from verified certificate chains in the
#   NSS certdb. Before bug 1619021, this would include preloaded intermediates,
#   which is unnecessary because cert_storage has a copy of those certificates, and
# ==============================================================================

diff -r d82b77ed1125 -r 889d7cd14e4d security/manager/ssl/nsNSSComponent.h
--- a/security/manager/ssl/nsNSSComponent.h	Fri Jun 05 00:39:30 2020 +0000
+++ b/security/manager/ssl/nsNSSComponent.h	Fri Jun 05 00:44:52 2020 +0000
@@ -29,7 +29,8 @@
 
 class nsIDOMWindow;
 class nsIPrompt;
-class SmartCardThreadList;
+class nsISerialEventTarget;
+class nsITimer;
 
 namespace mozilla {
 namespace psm {
@@ -107,6 +108,8 @@
 
   bool ShouldEnableEnterpriseRootsForFamilySafety(uint32_t familySafetyMode);
 
+  nsresult MaybeEnableIntermediatePreloadingHealer();
+
   // mLoadableCertsLoadedMonitor protects mLoadableCertsLoaded.
   mozilla::Monitor mLoadableCertsLoadedMonitor;
   bool mLoadableCertsLoaded;
@@ -136,6 +139,13 @@
   // to complete (because it will never complete) so we use this boolean to keep
   // track of if we should wait.
   bool mLoadLoadableCertsTaskDispatched;
+  // If the intermediate preloading healer is enabled, the following timer
+  // periodically dispatches events to the background task queue. Each of these
+  // events scans the NSS certdb for preloaded intermediates that are in
+  // cert_storage and thus can be removed. By default, the interval is 5
+  // minutes.
+  nsCOMPtr<nsISerialEventTarget> mIntermediatePreloadingHealerTaskQueue;
+  nsCOMPtr<nsITimer> mIntermediatePreloadingHealerTimer;
 };
 
 inline nsresult BlockUntilLoadableCertsLoaded() {